third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -1676,6 +1714,13 @@ impl Value {
         Self(ValueImpl::Container(Container::signer(x)))
     }
 
+    pub fn permissioned_signer(x: AccountAddress, perm_storage_address: AccountAddress) -> Self {
+        Self::struct_(Struct::pack_variant(PERMISSIONED_SIGNER_VARIANT, vec![
+            Value::address(x),
+            Value::address(perm_storage_address),
+        ]))
+    }
+
     /// Create a "unowned" reference to a signer value (&signer) for populating the &signer in
     /// execute function
     pub fn signer_reference(x: AccountAddress) -> Self {
```

Comment:
Can we rename `signer_reference` and `signer` to `master_signer_reference` and `master_signer`? Then it is clear that we do not create a permissioned one